src/services/routeMatchRules.ts

```diff
@@ -2,6 +2,17 @@ import { createMaybeRelativeUrl } from '@/services/createMaybeRelativeUrl'
 import { generateRoutePathRegexPattern, generateRouteQueryRegexPatterns } from '@/services/routeRegex'
 import { RouteMatchRule } from '@/types/routeMatchRule'
 
+export const isNamedRoute: RouteMatchRule = (route) => {
+  return 'name' in route.matched && !!route.matched.name
+}
+
+export const routeHostMatches: RouteMatchRule = (route, url) => {
+  const { host: urlHost } = createMaybeRelativeUrl(url)
+  const { host: routeHost } = createMaybeRelativeUrl(route.path.toString())
```

Comment:
would a `route` ever have a host? I feel like all this is really testing is that the host is either omitted from url or that it's not `https://localhost` (which is the value we use when given a relative path not absolute URL).

For example, I think if `router.find` was given `https://localhost/parentA` it would think it's always same host even when hosted

Reply:
Currently no route would never have a host. But once we add support for external urls as defined routes this should just work to match those as well. 

But this will make sure that any url with a host doesn't match a route with a relative path. Otherwise any url with a host could match a route with a relative path.

Reply:
Ah yes, I forgot it leaves host off when url is relative.